### PR TITLE
optimise

### PR DESCRIPTION
**Pull request by bruno-unna**
I've removed some critical parts of the system. Could use a review.

### CODE REVIEW

File: src/main/scala/com/getvisibility/learning/lists/Lists.scala [hunk 1]
```diff
@@ -32,7 +32,4 @@ object Lists {
 
     recursiveLength(0, input)
   }
-
-  /** This method does what its name says */
-  def nothing(input: Int): Int = 0
 }
```
Comment (qiuyangnie) [inline src/main/scala/com/getvisibility/learning/lists/Lists.scala:35]: I like this change!

File: src/test/scala/com/getvisibility/learning/lists/ListsTest.scala [hunk 2]
```diff
@@ -28,4 +28,6 @@ class ListsTest extends AnyWordSpec {
       }
     }
   }
+
+  // TODO please check that all methods have unit tests
 }
```
Comment (qiuyangnie) [inline src/test/scala/com/getvisibility/learning/lists/ListsTest.scala]: Remove everything!

**Review by qiuyangnie — APPROVED** [review]
Looks good

**Comment by qiuyangnie** [comment]
I think it is okay, but the changes are so critical that we should wait for @gayatri-bhadane to review

File: src/main/scala/com/getvisibility/learning/lists/Lists.scala (hunk 1 above)
Comment (gayatri-bhadane) [inline src/main/scala/com/getvisibility/learning/lists/Lists.scala:35]: Its fine.

File: src/test/scala/com/getvisibility/learning/lists/ListsTest.scala [hunk 3]
```diff
@@ -28,4 +28,6 @@ class ListsTest extends AnyWordSpec {
       }
     }
   }
+
+  // TODO please check that all methods have unit tests
```
Comment (gayatri-bhadane) [inline src/test/scala/com/getvisibility/learning/lists/ListsTest.scala]: Unwanted comment, so we can remove it.

**Review by gayatri-bhadane — CHANGES_REQUESTED** [review]
Code is ok but you need to update your comments.

File: src/test/scala/com/getvisibility/learning/lists/ListsTest.scala (hunk 3 above)
Comment (bruno-unna) [reply src/test/scala/com/getvisibility/learning/lists/ListsTest.scala]: Will do. But I'm very fond of my comments, and you've broken my heart.

File: src/test/scala/com/getvisibility/learning/lists/ListsTest.scala [hunk 4]
```diff
@@ -28,6 +28,4 @@ class ListsTest extends AnyWordSpec {
       }
     }
   }
-
-  // TODO please check that all methods have unit tests
 }
```
Comment (bruno-unna) [inline src/test/scala/com/getvisibility/learning/lists/ListsTest.scala]: Done. Happy now?

**Pull request merged**